lib/src/components/NxToast/__tests__/NxToastContainer.test.tsx

```diff
@@ -0,0 +1,127 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React from 'react';
+import { rtlRenderElement, rtlRender } from '../../../__testutils__/rtlUtils';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import NxToastContainer from '../NxToastContainer';
+import { NxToastContainerProps } from '../types';
+import NxToast from '../NxToast';
+
+import NxAlert from '../../NxAlert/NxAlert';
+import { faEye } from '@fortawesome/free-solid-svg-icons';
+
+describe('NxToastContainer', function() {
+  const simpleProps: NxToastContainerProps = {
+    children:
+  <>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+    <NxToast onClose={()=>{}}>
+      <NxAlert icon={faEye}>This is an Alert</NxAlert>
+    </NxToast>
+  </>
+  };
+
+  const quickRender = rtlRender(NxToastContainer, simpleProps);
+  const renderEl = rtlRenderElement(NxToastContainer, simpleProps);
+
+  it('renders any NxToasts passed into it', function() {
+    renderEl();
+
+    const alerts = screen.getAllByText('This is an Alert');
+
+    expect(alerts.length).toBe(3);
+  });
+
+  it('sets focus to last toast when first toast is closed', async function() {
+    const user = userEvent.setup();
+    quickRender();
+
+    const closeBtns = screen.getAllByRole('button', {name: 'Close'});
+
+    await user.click(closeBtns[0]);
+
+    expect(closeBtns[2]).toHaveFocus();
+  });
+
+  it('sets focus to the next last toast when the last toast is closed', async function() {
+    const user = userEvent.setup();
+    quickRender();
+
+    const closeBtns = screen.getAllByRole('button', {name: 'Close'});
+
+    await user.click(closeBtns[2]);
+
+    expect(closeBtns[1]).toHaveFocus();
+  });
+
+  it('sets focus to previous focused element when the last remaining toast is closed',
+      async function() {
+        const Component = () => {
+          return (
+            <>
+              <button type="button">Focus Me</button>
+              <NxToastContainer>
+                <NxToast onClose={()=>{}}>
+                  <NxAlert icon={faEye}>This is an Alert</NxAlert>
+                </NxToast>
+              </NxToastContainer>
+            </>
+          );
+        };
+
+        render(<Component />);
+
+        const prevFocusBtn = screen.getByRole('button', {name: 'Focus Me'});
+        const closeBtn = screen.getByRole('button', {name: 'Close'});
+
+        prevFocusBtn.focus();
+
+        await userEvent.click(closeBtn);
+
+        expect(prevFocusBtn).toHaveFocus();
+      });
+
+  it('sets focus to previous focused element (while toasts are rendered) when the last remaining toast is closed',
```

Comment:
I'm not sure I understand the intent of having this test in addition to the one above

Reply:
> I'm not sure I understand the intent of having this test in addition to the one above

Perahps I misunderstood, but I was thinking about your comment in https://github.com/sonatype/sonatype-react-shared-components/pull/688#discussion_r953913196, although reflecting on it now it seems like both do the same thing, just in a different order. I can remove this test if you'd like.

Reply:
> Perahps I misunderstood, but I was thinking about your comment in https://github.com/sonatype/sonatype-react-shared-components/pull/688#discussion_r953913196, although reflecting on it now it seems like both do the same thing, just in a different order. I can remove this test if you'd like.

Ok I see what you're going for. As written though I think these tests are essentially the same.  Instead, I think what we want is one test where `prevFocusBtn` gets focused after the toast opens (as these tests have), and another test where `preFocusBtn` was only focused _before_ the toast was opened, and then the toast is closed with no other element having been focused after the toast was opened

Reply:
> Ok I see what you're going for. As written though I think these tests are essentially the same. Instead, I think what we want is one test where `prevFocusBtn` gets focused after the toast opens (as these tests have), and another test where `preFocusBtn` was only focused _before_ the toast was opened, and then the toast is closed with no other element having been focused after the toast was opened

So I've updated the final test here: [7bc19b1](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/7bc19b1fcae9c3a2152c8d3e7abbb43bac641967). I'm not sure why, but with the rerender, the test seems to need the animationEnd event to fire in order to wait for the focus to be set back to the prevFocusBtn (while the other tests did not).

Reply:
> I'm not sure why, but with the rerender, the test seems to need the animationEnd event to fire in order to wait for the focus to be set back to the prevFocusBtn (while the other tests did not).

Are you sure? I just tried commenting out the `fireEvent.animationEnd` (and resulting unused variables) and the test passed.

Reply:
> Are you sure? I just tried commenting out the `fireEvent.animationEnd` (and resulting unused variables) and the test passed.

Odd, but it's passing on my end now as well. I must have been doing something wrong. It's been changed here: [c158bb7](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/c158bb70dd6e54f0022e400d5ee0b1706b214adc)